homeassistant/components/otbr/util.py

```diff
@@ -68,6 +71,17 @@ class OTBRData:
     api: python_otbr_api.OTBR
     entry_id: str
 
+    @_handle_otbr_error
+    async def factory_reset(self) -> None:
+        """Reset the router."""
+        try:
+            await self.api.factory_reset()
+        except python_otbr_api.FactoryResetNotSupportedError:
+            _LOGGER.info(
```

Comment:
I think I'd make this even a warning. Only deleting the dataset leaves the BR with the same Border Agent ID, and the Thread network seems to still carry old IPv6 addresses. We should really encourage users to upgrade to newer versions of the add-on which do a proper reset.

```suggestion
            _LOGGER.warning(
```

(Note that the latest OpenThread Border Router add-on and the latest Silicon Labs Multiprotocol add-on have the proper factory reset and won't show this warning).